gnomad/utils/constraint.py

```diff
@@ -125,3 +136,251 @@ def collapse_strand(
+        - y: sum('observed_variants')/ (`high_coverage_scale_factor` * sum('possible_variants' * 'mu_snp') at low coverage sites
+
+        `high_coverage_scale_factor` = sum('observed_variants') / sum('possible_variants' * 'mu_snp') at high coverage sites
+
+
+    .. note::
+
+        This function expects that the input Table(`coverage_ht`) was created using
+        `get_proportion_observed_by_coverage`, which means that `coverage_ht` should contain only high quality
+        synonymous variants below 0.1% frequency.
+
+        This function also expects that the following fields are present in `coverage_ht`:
+        - context - trinucleotide genomic context
+        - ref - the reference allele
+        - alt - the alternate base
+        - methylation_level - methylation level
+        - cpg - whether the site is CpG site
+        - exome_coverage - median exome coverage at integer values between 1-100
+        - observed_variants - the number of observed variants in the dataset for each variant. Note that the term
+        "variant" here refers to a specific substitution, context, methylation level, and coverage combination.
+        - downsampling_counts_{pop} (optional) - array of observed variant counts per population after downsampling when `pops` is specified.
+        - mu_snp - mutation rate
+        - possible_variants - the number of possible variants in the dataset for each variant.
+
+    :param coverage_ht: Input coverage Table.
+    :param weighted: Whether to weight the high coverage model (a linear regression model) by 'possible_variants'. Defaults to False.
+    :param pops: List of populations used to build coverage and plateau models. Defaults to ().
+    :param keys: Keys to group observed and possible variant counts.
+        Default is ("context","ref", "alt", "methylation_level", "mu_snp").
+    :param cov_cutoff: Median coverage cutoff. Sites with coverage above this cutoff are considered well covered
+    and will be used to build plateau models. Sites below this cutoff have low coverage and will be used to build
+    coverage models. Defaults to `COVERAGE_CUTOFF`.
+    :return: Coverage model and plateau models.
+    """
+    # Filter to sites with coverage above `cov_cutoff`
+    all_high_coverage_ht = coverage_ht.filter(coverage_ht.exome_coverage >= cov_cutoff)
+    agg_expr = {
+        "observed_variants": hl.agg.sum(all_high_coverage_ht.observed_variants),
+        "possible_variants": hl.agg.sum(all_high_coverage_ht.possible_variants),
+    }
+    for pop in pops:
+        agg_expr[f"observed_{pop}"] = hl.agg.array_sum(
+            all_high_coverage_ht[f"downsampling_counts_{pop}"]
+        )
+
+    # Generate a Table with all necessary annotations (x and y listed above) for the plateau models
+    high_coverage_ht = all_high_coverage_ht.group_by(*keys).aggregate(**agg_expr)
+    high_coverage_ht = annotate_mutation_type(high_coverage_ht)
+
+    # Build plateau models
```

Comment:
Do we want to check the length of observed variant counts array for each population here? The check can't be placed check to `build_plateau_models()` because it needs `ht` for aggregation. The only concern is calling `get_all_pop_lengths()` may be redundant, because the dictionary it returns won't be used. Please let me know if you have any ideas to simplify the code!